src/test/java/com/rocketden/main/service/GameManagementServiceTests.java

```diff
@@ -263,6 +264,94 @@ public void submitSolutionEmptyField() {
         assertEquals(GameError.EMPTY_FIELD, exception.getError());
     }
 
+    @Test
+    public void playAgainSuccess() throws Exception {
+        User host = new User();
+        host.setNickname(NICKNAME);
+        host.setUserId(USER_ID);
+        host.setSessionId(SESSION_ID);
+
+        Room room = new Room();
+        room.setRoomId(ROOM_ID);
+        room.setHost(host);
+        room.setDifficulty(ProblemDifficulty.HARD);
+        room.setActive(true);
+        room.addUser(host);
+        room.setDuration(1L);
+
+        StartGameRequest request = new StartGameRequest();
+        request.setInitiator(UserMapper.toDto(host));
+
+        Mockito.doReturn(room).when(repository).findRoomByRoomId(ROOM_ID);
+        gameService.startGame(ROOM_ID, request);
+        Thread.sleep(1000);
```

Comment:
nit: Replace `Thread.sleep` with a different testing implementation (in all instances, there is another below for example):

<img width="940" alt="Screen Shot 2021-01-23 at 1 40 57 PM" src="https://user-images.githubusercontent.com/7987279/105614777-741a2680-5d80-11eb-9707-a98fbc7d106b.png">

For the `verify` check, I believe you can do `verify(socketService, timeout(1000)).sendSocketUpdate(eq(gameDto));` -- I do this on `EndGameTimerTaskTests.java`.

Reply:
Sounds good